README.md

```diff
@@ -1 +1,1181 @@
+            match msg {
+                json jsonPL => {
+                    string message;
+                    if (lengthof jsonPL >= 1) {
+                        // Validate to check if student with given ID exist in the system.
+                        message = "Maths: " + jsonPL[0]["maths"] .toString() + " English: " + jsonPL[0]["english"] .toString() + " Science: " + jsonPL[0]["science"] .toString();
+                    }
+                    else {
+                        message = "Data not available. Check if student's mark is added or student might not be in our system.";
+                    }
+                    io:println("\n"+ message + "\n");
+                }
+                error err => {
+                    log:printError(err.message, err = err);
+                }
+            }
+        }
+        error err => {
+            log:printError(err.message, err = err);
+        }
+    }
+}
+```
+
+- Now we have completed the implementation of student management service with marks management service.
+
+
+## Testing 
+
+### Invoking the student management service
+
+You can start both the services by opening a terminal and navigating to `ballerina-honeycomb/guide`, and execute the following command.
+
+```
+$ ballerina run --config <path-to-conf>/ballerina.conf students
+```
+
+ You need to start the honeycomb-opentracing-proxy. This can be done by using docker. Docker is used to pull the image for honeycomb-opentracing-proxy.
+ Run the following command.
+ 
+ ```
+ docker run -p 9411:9411 honeycombio/honeycomb-opentracing-proxy -k APIKEY -d traces
+```
+- -k represent the API KEY you will be getting when you sign in to honeycomb account.
+
+- -d represents the dataset you are going to send your trace data to.
+
+ You can observe the service performance by making some HTTP requests to the above services. This is made easy for you as 
+ there is a client program implemented. You can start the client program by opening another terminal and navigating to ballerina-honeycomb/guide
+ and run the below command
+ 
+ ```
+ $ ballerina run client_service
+ ``` 
+ 
+### Testing with Honeycomb
+ 
+#### Views of traces
+ After making http request, go to [Honeycomb website](https://honeycomb.io) then move to your dataset.
+
+   When you are in your dataset in Honeycomb UI you get to see a button called `New query`, and when you click on that 
+   you can write your own queries on the metrics that you have received.
+ - You are expected to see the traces as below when you include traceId in the breakdown category.
+ 
+![Honeycomb](images/traces1.png "Honeycomb")
+ 
+ - To view a particular trace click on the traceId column. And you will see as below
+ 
+![Honeycomb](images/traces2.png "Honeycomb")
+    
+ - To view span details with metrics click on a particular span and you are expected to see as below
+ 
+![Honeycomb](images/traces3.png "Honeycomb")
+     
+     
+#### Metrics
```

Comment:
The formatting for this entire section needs some improvements. There are extra tab spaces, issues with the positioning of numbered lists, and sentences with extra line breaks. Please check and fix.